### DISCUSSION TITLE
Advice on how to obtain credentials on Google Cloud Run

### DISCUSSION THREAD

**Discussion by atrauzzi** [stackkit/laravel-google-cloud-tasks-queue#21] [Q&A] [answered]
Hey there!

I'm just going through getting set up with this package.  I am somewhat new to Cloud Run so I'm not 100% familiar with what's defined in the environment or what will and what won't be picked up automatically.

Do I have to explicitly set `GOOGLE_APPLICATION_CREDENTIALS` for this library to authenticate correctly, or will things _just work_ :tm: by way of the fact that it's running on Google Cloud Run? I only ask this because I'm not sure how or what to generate for the environment variable otherwise.  The documentation linked to [explains automatic credential discovery](https://cloud.google.com/docs/authentication/production#automatically), which would definitely be more convenient than having another environment variable to manage.

Thanks!

cc. @marickvantuil

**Comment by marickvantuil** [comment, marked as answer]
Hey, first of all, my bad, I somehow missed this message. When running an application from Cloud Run, it will automatically have the correct `GOOGLE_APPLICATION_CREDENTIALS` environment configured. But if not, you can set it manually and reference the path to the JSON key file which can be created for a service account. More info about that here: https://cloud.google.com/docs/authentication/getting-started

Hope that helps.
